### PR TITLE
Adapts acceptance test suite to run against Private Triton installations

### PR DESCRIPTION
**Pull request by joshado**
I've been working on adding `network_object` specifiers to this provider and, as part of this, I've needed to run the acceptance tests against our private Triton deployment. This was non-trivial and lead to a number of patches overriding package and network names.

This PR adapts the test suite to pull in those arguments from the environment - defaulting the values to be compatible with the Joyent Public Cloud.

The specific changes I've made to achieve this are:

* Bumps the `triton-go` dependency to `v1.7.0`
* Introduces a testAccConfig(t, key) method to retrieve private triton deployment specific configuration by name from the environment
* Use the `triton.GetEnv(..)` method in Acceptance test rather than querying the environment so behaviour is consistent with provider  re. SDC_ and TRITON_ prefix.
* Removes the “OldURL” test and substitution as this is public-cloud specific, which is being retired
* Keep test VLAN numbers high (>1024) to avoid potential VLAN collisions
* Add a `public_network_name` parameter replacing static `Joyent-SDC-Public`
* Add a `test_package_name` and `test_network_name` to allow these to be changed.
* Adds a set of parameters to test package lookup
* A heap of structural changes to fixtures in resource_machine_test:
  - explicitly lookup and specify the network when provisioning VMs rather than relying on the CloudAPI default
  - coalesce test fixtures into a set of functions to avoid unnecessary repetition
  - parametrise the network name and package name
* Change the snapshot tests to use the same base image as other machine tests
* Documents the parameter key names in the testing section of the `README.md`

### CODE REVIEW

**Comment by siepkes** [comment]
Perhaps it's an idea (probably for some future PR) to create a script which configures a cleanly installed COAL instance as a baseline for acceptance tests (vlan's, packages, etc.). 

A while back I created a [script](https://github.com/joyent/triton/blob/master/tools/coal-linux-kvm-setup) for using COAL with libvirt / KVM instead of vmware. I guess we could look into extending that?

**Comment by joshado** [comment]
I wondered about automating the tests against CoaL - would be nice, but even just having the parameters in one place now makes it much easier to do this work as a next-step. It did occur to me that (given a big enough machine) you could virtualise a small CoaL cluster and run the tests automatically.

I've just pushed some formatting fixes - hopefully Travis will be happy now -  are there any other blockers to getting this merged? Is there anything else I'd need to do?

File: triton/resource_machine_test.go [hunk 1]
```diff
@@ -470,483 +515,273 @@ func TestAccTritonMachine_deletionProtection(t *testing.T) {
...
+		  name = "%s-3"
+		  package = "%s"
+		  image = "${data.triton_image.base.id}"
 
-resource "triton_machine" "test1" {
-  name = "%s-1"
-  package = "g4-highcpu-128M"
-  image = "${data.triton_image.base.id}"
-}
+			networks = [data.triton_network.test.id]
```
Comment (twhiteman) [inline triton/resource_machine_test.go]: Indentation is inconsistent.

File: triton/data_source_fabric_network_test.go [hunk 2]
```diff
@@ -55,7 +55,7 @@ func TestAccTritonFabricNetwork_NotFound(t *testing.T) {
 }
 
 func TestAccTritonFabricNetwork_Basic(t *testing.T) {
-	vlanID := acctest.RandIntRange(3, 2048)
+	vlanID := acctest.RandIntRange(1024, 2048)
```
[the remark this answers is not in the document]
Comment (joshado) [reply triton/data_source_fabric_network_test.go]: Possibly - I wanted to limit the scope of this PR just to allow me to run the tests against a private triton instance. It does still leave a heap of Fabric VLANs around - which isn't ideal - and also I occasionally saw failures in the `triton_key` complaining that "Key xxx Already exists".

File: triton/provider_test.go [hunk 3]
```diff
@@ -42,3 +45,41 @@ func testAccPreCheck(t *testing.T) {
...
+	}
+
+	var env_value = os.Getenv(fmt.Sprintf("testacc_%s", key))
+	if env_value != "" {
+		return env_value
+	}
+
+	switch key {
+	case "dc_name":
+		return "us-sw-1"
```
[the remark this answers is not in the document]
Comment (joshado) [reply triton/provider_test.go]: Ah bother, I accidentally copied the wrong one!

File: triton/resource_machine_test.go [hunk 4]
```diff
@@ -302,26 +302,23 @@ func TestAccTritonMachine_firewall(t *testing.T) {
 
 func TestAccTritonMachine_metadata(t *testing.T) {
 	machineName := fmt.Sprintf("acctest-%d", acctest.RandInt())
-	basic := fmt.Sprintf(testAccTritonMachine_metadata_1, machineName)
-	add_metadata := fmt.Sprintf(testAccTritonMachine_metadata_1, machineName)
-	add_metadata_2 := fmt.Sprintf(testAccTritonMachine_metadata_2, machineName)
-	add_metadata_3 := fmt.Sprintf(testAccTritonMachine_metadata_3, machineName)
-	add_metadata_4 := fmt.Sprintf(testAccTritonMachine_metadata_4, machineName)
-	add_metadata_5 := fmt.Sprintf(testAccTritonMachine_metadata_5, machineName)
+	// testAccTriton_metadata(t, <name>, <outer prepend>, <machine append>)
```
[the remark this answers is not in the document]
Comment (joshado) [reply triton/resource_machine_test.go]: Ignore this - it was a reminder for me. Removed.

File: triton/resource_machine_test.go [hunk 5]
```diff
@@ -357,7 +379,17 @@ func TestAccTritonMachine_metadata(t *testing.T) {
 				),
 			},
 			{
-				Config: add_metadata_5,
+				Config: testAccTritonMachine_metadata(t, machineName, "", `
+					user_data = "hello"
```
[the remark this answers is not in the document]
Comment (joshado) [reply triton/resource_machine_test.go]: :+1:

File: triton/data_source_fabric_network_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (twhiteman) [reply triton/data_source_fabric_network_test.go]: Sounds good for separate work

**Pull request merged**